### PR TITLE
Featureimage/new snmpv3 encryption

### PR DESCRIPTION
**Pull request by stark-dev**

### CODE REVIEW

**Comment by stark-dev** [comment]
Please DO NOT MERGE until we have PPO validation

File: lib/CMakeLists.txt [hunk 1]
```diff
@@ -74,6 +74,7 @@ etn_target(shared ${PROJECT_NAME_UNDERSCORE}
         fty_common_messagebus
         fty_common_dto
         fty_lib_certificate
+        fty-utils
```
Comment (etnKlendathu) [inline lib/CMakeLists.txt:77]: did you add utils only for convert in version check?

File: lib/src/secw_security_wallet.cc [hunk 2]
```diff
@@ -185,7 +186,7 @@ cxxtools::SerializationInfo SecurityWallet::getSrrSaveData(const std::string& pa
 void SecurityWallet::restoreSRRData(
     const cxxtools::SerializationInfo& si, const std::string& passphrase, const std::string& version)
 {
-    if (version != "1.0") {
+    if (fty::convert<float>(version) > fty::convert<float>(ACTIVE_VERSION)) {
```
Comment (etnKlendathu) [inline lib/src/secw_security_wallet.cc:189]: I did not understand why ACTIVE_VERSION, do we have INACTIVE_VERSION? :)

File: lib/tests/producer_accessor.cpp [hunk 3]
```diff
@@ -500,6 +500,177 @@ void secwProducerAccessorTest(fty::SocketSyncClient& syncClient, mlm::MlmStreamC
         }
     }
 
+    // test 5.7 => insertNewDocument SNMPV3 - SHA256 / AES256
```
Comment (etnKlendathu) [inline lib/tests/producer_accessor.cpp:503]: Please, god, make someone to change this unit testing :)

**Review by etnKlendathu — APPROVED** [review]

**Pull request closed without merging**